src/lib.rs

```diff
@@ -327,12 +353,6 @@ impl<T: AsRef<[u8]>> TOTP<T> {
             }
         }
 
-        if issuer.is_some() && issuer.as_ref().unwrap().contains(':') {
-            return Err(TotpUrlError::Issuer);
-        }
-        if account_name.contains(':') {
-            return Err(TotpUrlError::AccountName);
-        }
```

Comment:
Removed these checks because they are redundant with the return of `TOTP::new`

Reply:
LGTM